### PR TITLE
Add pr workflow

### PR DESCRIPTION
**Pull request by JamesGardiner**
- pull_request.yaml (run on pull requests to master branch)
  - publish_to_pypi.yaml (run on a release)

## Checklist
  - [x] CHANGELOG.md updated? (if required)

### CODE REVIEW

**Review by pricem14pc — COMMENTED** [review]
I guess the Travis and Codecov badges can come out of the README now

File: setup.py [hunk 1]
```diff
@@ -13,13 +13,19 @@
...
+                "r",
+            )
+            .read()
+            .split("=")[-1]
+            .strip()
         )
     )
 
+with open("./README.md") as fp:
+    description = fp.read()
```
Comment (pricem14pc) [inline setup.py:27]: This does seem to work but I'm not sure how relevant the info is in the Pypi package long description.

Comment (pricem14pc) [reply setup.py:27]: In lieu of another long description and the fact it needs a value to suppress the warning it'll probably do. There are other examples in Pypi where the README is simply used here.

**Comment by warrenbailey** [comment]
LGTM 👍

**Review by warrenbailey — APPROVED** [review]

**Review by pricem14pc — APPROVED** [review]

**Pull request merged**